state/core_access.go

```diff
@@ -157,6 +175,11 @@ func (ca *CoreAccessor) Start(ctx context.Context) error {
 	}
 	ca.rpcCli = cli
 
+	err = ca.setupSigner(ctx)
+	if err != nil {
+		log.Warnw("failed to set up signer, check if node's account is funded", "err", err)
+	}
```

Comment:
[blocking]
Why isn't this a hard error?

I can see that `getSigner` can also construct signer lazily, but:
* It's racy - there is no synchronization around the signer struct
* There is no obvious reason to attempt to construct a Signer on every TX if signer construction fails on Start.

Reply:
@Wondertan yes it's a bit racey so I can put a mutex there for additional safety but the reason I decided to go for the lazy construction is because it should be possible for a node runner to run a node first, get the address of the new key that's generated, and then while the node is still running, go fund the account, then use state module once the account is funded. 

Signer is unable to be constructed before the account "exists" which is why I opted for lazy construction.

Reply:
Ok, would be great to have this nuance on the pr description.

Reply:
https://github.com/celestiaorg/celestia-node/pull/3298/commits/24b8b2ff28b4eed8eac9fb3539cc116562a30803

Reply:
Let's add the comment, though. It's a detail that is easy to miss, and the lazy construction should be justified for anyone looking at the code in future.

Reply:
done